### DISCUSSION TITLE
start server an error

### DISCUSSION THREAD

**Discussion by bouyeijiang** [dotnet/orleans#7894] [Q&A]
host = new HostBuilder()
            .UseOrleans(builder =>
            {
                builder.Configure<ClusterOptions>(options =>
                {
                    options.ClusterId = "dev";
                    options.ServiceId = "dev";
                }).ConfigureEndpoints(siloPort: 11111, gatewayPort: 30000)
                    .ConfigureApplicationParts(
                        parts => parts.AddApplicationPart(typeof(GrainGdbQuery).Assembly).WithReferences()
                        );
            }).Build();

            await host.StartAsync();

exception:
System.InvalidOperationException:“Unable to resolve service for type 'Orleans.IMembershipTable' while attempting to activate 'Orleans.Runtime.MembershipService.MembershipTableManager'.”

**Comment by JorgeCandeias** [comment]
That code is missing a clustering provider, the docs [here](https://docs.microsoft.com/en-us/dotnet/orleans/host/configuration-guide/local-development-configuration) show a correct example.

**Comment by bouyeijiang** [comment]
i have solved the problem,according do the example and i found have to import microsoft.orleans.codegenerator.msbuild.dll
.thinks

**Reply by JorgeCandeias** [reply]
If you're learning Orleans from scratch, there's now an intro on Microsoft Learn that is more accessible than the docs:

https://docs.microsoft.com/en-us/learn/modules/orleans-build-your-first-app/
